ideascube/mixins.py

```diff
@@ -72,10 +74,10 @@ def _set_available_tags(self, context):
 
     def get_context_data(self, **kwargs):
         context = super(FilterableViewMixin, self).get_context_data(**kwargs)
-        for key in ('q', 'kind', 'lang'):
+        for key in ('q', 'kind', 'lang', 'source'):
             context[key] = self.request.GET.get(key)
         context['tags'] = self.request.GET.getlist('tags')
-        current_filters = [(k, context[k]) for k in ('kind', 'lang')
+        current_filters = [(k, context[k]) for k in ('kind', 'lang', 'source')
```

Comment:
Agree with your general argument to try to remove unnecessary loops.

In this particular case, we are looping over 3 items in a list comprehension without any complex call (sql request, ...). I have no idea about how it competes with a extra `if` in a classical loop but I pretty sure the difference is negligible :)